jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/commons/JavaCursorContextKind.java

```diff
@@ -226,8 +226,15 @@ public int getValue() {
 
     public static JavaCursorContextKind forValue(int value) {
         JavaCursorContextKind[] allValues = JavaCursorContextKind.values();
-        if (value < 1 || value > allValues.length)
-            throw new IllegalArgumentException("Illegal enum value: " + value);
-        return allValues[value - 1];
+		
+        // Root cause and fix for https://github.com/eclipse-lsp4jakarta/lsp4jakarta/issues/520.
```

Comment:
Please update the comment to describe the reason for the if stmt rather than refer to the issue number. This makes it easier for someone else to understand the code because they do not need to go look up the issue.